avalon/nuke/lib.py

```diff
@@ -3,82 +3,283 @@
 import nuke
 import re
 import logging
-import toml
+import six
+from ..vendor import clique
 
 log = logging.getLogger(__name__)
 
 
 @contextlib.contextmanager
 def maintained_selection():
+    """Maintain selection during context
+
+    Example:
+        >>> with maintained_selection():
+        ...     node['selected'].setValue(True)
+        >>> print(node['selected'].value())
+        False
+    """
+    nodes = nuke.allNodes()
     previous_selection = nuke.selectedNodes()
     try:
         yield
     finally:
         # unselect all selection in case there is some
-        current_seletion = nuke.selectedNodes()
-        [n['selected'].setValue(False) for n in current_seletion]
+        [n['selected'].setValue(False) for n in nodes]
         # and select all previously selected nodes
         if previous_selection:
-            [n['selected'].setValue(True) for n in previous_selection]
+            [n['selected'].setValue(True)
+             for n in nodes
+             if n.name() in previous_selection]
+
+
+def reset_selection():
+    """Deselect all selected nodes
+    """
+    nodes = nuke.allNodes()
+    [n['selected'].setValue(False) for n in nodes]
```

Comment:
See [this comment](https://github.com/getavalon/core/pull/438/files#r319584885) about not using list comprehensions to perform actions, it's not pythonic.

Plus, for deselecting you wouldn't need to iterate over all nodes. Instead just do:
```python
for node in nuke.selectedNodes():
    node['selected'] = False
```

Reply:
If there's another reason to use `allNodes()` then note that Nuke also seems to expose [`clear_selection_recursive` in nukescripts](https://learn.foundry.com/nuke/developers/70/pythonreference/nukescripts.misc-pysrc.html#clear_selection_recursive). Not sure if that helps.